### PR TITLE
Fix queue view to show provider label instead of raw ID

### LINKED ISSUE #227: Queue view shows raw provider ID instead of human-readable label
## Problem

In the Queue tree view, provider-backed items show the raw `providerId` string (e.g., `"github"`) as the tree item description. The `ProviderRegistry.getProviderLabel()` method exists and returns the human-readable label, but it is not used in `QueueTreeProvider.createWorkItemTreeItem()`.

```typescript
// Current (queueTreeProvider.ts:39)
treeItem.description = item.providerId;

// Should use the provider label resolver instead
```

## Expected Behavior

The Queue view should display the human-readable provider label (e.g., "GitHub Issues") instead of the raw provider ID. The Focus and History views already have access to provider label resolution via their constructor — the Queue view should follow the same pattern.

### PR DESCRIPTION
**Pull request by mthalman**
Fix queue view to show provider label instead of raw ID

Closes #227

## Problem

The Queue tree view displayed the raw `providerId` string (e.g., `"github"`) as the tree item description. The `ProviderRegistry.getProviderLabel()` method exists and returns the human-readable label, but `QueueTreeProvider` wasn't using it.

## Solution

Added a `getProviderLabel()` protected method to the base `WorkItemViewProvider` class and updated `QueueTreeProvider` to use it. This follows the same pattern already used by `FocusTreeProvider` and `HistoryTreeProvider` for label resolution.

Queue items now display "GitHub Issues" instead of "github", "Azure DevOps" instead of "ado", etc.

## Changes

- `packages/core/src/views/viewLayout.ts` — Added `getProviderLabel()` to base class
- `packages/core/src/views/queueTreeProvider.ts` — Use resolved label instead of raw ID
- `packages/core/src/test/queueTreeProvider.test.ts` — 6 new tests for label resolution

## Testing

All 870 core tests pass (including 6 new). Full suite (1298 tests) green.

### CODE REVIEW

File: packages/core/src/views/viewLayout.ts [hunk 1]
```diff
@@ -318,6 +318,10 @@ export abstract class WorkItemViewProvider implements vscode.TreeDataProvider<Wo
 
   refresh(): void { this._onDidChangeTreeData.fire(); }
 
+  protected getProviderLabel(providerId: string | undefined): string | undefined {
+    return providerId && this.labelResolver ? this.labelResolver(providerId) : providerId;
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/src/views/viewLayout.ts]: Fixed in bd1852c

File: .squad/agents/fenster/history.md [hunk 2]
```diff
@@ -239,6 +239,26 @@ Patterns documented in `.squad/decisions.md` under "Code Review Fix Patterns" (2
...
+## Issue #227: Queue View Provider Labels (2026-04-13)
+
+**Status:** COMPLETE — Provider labels now display in queue view instead of raw IDs
+
+### Summary
+The queue view was displaying raw provider IDs (e.g., `github`, `ado`) in tree items. Extracted `getProviderLabel()` method from `WorkItemViewProvider` base class and applied it in `QueueTreeProvider` to show human-readable labels (e.g., "GitHub", "Azure DevOps").
+
+### Files Modified
+- `packages/core/src/views/queueTreeProvider.ts` — Updated tree item label rendering
+- `packages/core/src/views/baseWorkItemViewProvider.ts` — Extracted `getProviderLabel()` method for reuse across all views
```
[the remark this answers is not in the document]
Comment (mthalman) [reply .squad/agents/fenster/history.md]: Fixed in 46a9009

File: packages/core/src/test/queueTreeProvider.test.ts [hunk 3]
```diff
@@ -426,4 +439,64 @@ describe('QueueTreeProvider', () => {
...
+    });
+
+    it('should return undefined if resolver returns undefined', async () => {
+      (providerRegistry.getProviderLabel as ReturnType<typeof vi.fn>).mockReturnValue(undefined);
+      const item = await graph.createItem(
+        { title: 'Unknown provider' },
+        { providerId: 'unknown', externalId: 'ext-1' },
+      );
+      const treeItem = providerWithResolver.getTreeItem(item);
+      expect(treeItem.description).toBeUndefined();
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/src/test/queueTreeProvider.test.ts]: Fixed in f0174a5

File: packages/core/src/views/viewLayout.ts [hunk 4]
```diff
@@ -318,6 +318,13 @@ export abstract class WorkItemViewProvider implements vscode.TreeDataProvider<Wo
 
   refresh(): void { this._onDidChangeTreeData.fire(); }
 
+  protected getProviderLabel(providerId: string | undefined): string | undefined {
+    if (!providerId) {
+      return providerId;
+    }
+    return this.labelResolver?.(providerId) ?? providerId;
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/src/views/viewLayout.ts]: Fixed in 9f42a50

File: .squad/agents/fenster/history.md [hunk 5]
```diff
@@ -239,6 +239,26 @@ Patterns documented in `.squad/decisions.md` under "Code Review Fix Patterns" (2
 - **Defensive async loading**: When a synchronous getter reads from an async-loaded cache, ensure the cache is loaded at every call site. Don't rely solely on initialization order guarantees.
 - **Test timing**: When production code adds an async operation to a previously-synchronous code path, tests that fire-and-forget events need to wait for async handlers to complete before asserting on side effects.
 
+## Issue #227: Queue View Provider Labels (2026-04-13)
+
+**Status:** COMPLETE — Provider labels now display in queue view instead of raw IDs
+
+### Summary
+The queue view was displaying raw provider IDs (e.g., `github`, `ado`) in tree items. Extracted `getProviderLabel()` method from `WorkItemViewProvider` base class and applied it in `QueueTreeProvider` to show human-readable labels (e.g., "GitHub", "Azure DevOps").
```
[the remark this answers is not in the document]
Comment (mthalman) [reply .squad/agents/fenster/history.md]: Fixed in c6fc9ec

File: .squad/agents/hockney/history.md [hunk 6]
```diff
@@ -373,4 +373,34 @@ mockFetch.mockImplementation(async (url: string) => {
...
+1. Queue tree items display correct provider labels
+2. Label lookup falls back gracefully when provider not found
+3. Label formatting and display consistency
+4. Integration with `getProviderLabel()` method from base provider
+5. Multiple provider types (GitHub, ADO) correctly labeled
+6. Edge case: items with missing provider IDs
+
+### Test Infrastructure Notes
+- Used existing `createMockProvider()` and `createMockStore()` patterns from Phase 2
+- No new test infrastructure required
```
[the remark this answers is not in the document]
Comment (mthalman) [reply .squad/agents/hockney/history.md]: Fixed in fc5044c
